gptel-context.el

```diff
@@ -163,32 +165,57 @@ context chunk.  This is accessible as, for example:
         (eq buffer-file-coding-system 'no-conversion))
     (file-missing (message "File \"%s\" is not readable." path))))
 
+(defun gptel-context--handle-binary (path)
+  "Add binary file at PATH to context if supported.
+Return PATH if added, nil if ignored."
+  (if-let* (((gptel--model-capable-p 'media))
+            (mime (mailcap-file-name-to-mime-type path))
+            ((gptel--model-mime-capable-p mime)))
+      (prog1 path
+        (cl-pushnew (list path :mime mime)
+                    gptel-context--alist :test #'equal)
+        (message "File \"%s\" added to context." path))
+    (message "Ignoring unsupported binary file \"%s\"." path)
+    nil))
+
+(defun gptel-context--handle-directory (path action)
+  "Process all files in directory at PATH according to ACTION.
+ACTION should be either `add' or `remove'."
+  (let ((files (directory-files-recursively path "." t)))
+    (mapc (lambda (file)
+            (unless (file-directory-p file)
+              (pcase action
+		('add (gptel--file-binary-p file)
+		      (gptel-context--handle-binary file)
```

Comment:
Am I missing something here?  Seems like this should be `(if (gptel--file-binary-p file) (gptel-context--handle-binary file) ...`.

Reply:
Yep, not sure what happened. Will make these and the other changes shortly.

Reply:
Addressed: https://github.com/karthink/gptel/pull/438/commits/2a40daef00c91620f142c0df8494883cdf3c66fe